boards/arm/waveshare_open103z/waveshare_open103z.dts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2020 Stefano Manni
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <st/f1/stm32f103Xe.dtsi>
+
+/ {
+	model = "Waveshare Open103Z";
+	compatible = "waveshare,open103z", "st,stm32f103ze";
+
+	chosen {
+		zephyr,console = &usart1;
+		zephyr,shell-uart = &usart1;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,can-primary = &can1;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		led_1: led_1 {
+			gpios = <&gpiof 6 GPIO_ACTIVE_HIGH>;
+			label = "User LED1";
+		};
+		led_2: led_2 {
+			gpios = <&gpiof 7 GPIO_ACTIVE_HIGH>;
+			label = "User LED2";
+		};
+		led_3: led_3 {
+			gpios = <&gpiof 8 GPIO_ACTIVE_HIGH>;
+			label = "User LED3";
+		};
+		led_4: led_4 {
+			gpios = <&gpiof 9 GPIO_ACTIVE_HIGH>;
+			label = "User LED4";
+		};
+	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+		button: button {
+			label = "User button";
+			gpios = <&gpiog 6 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+		};
+		joystick_enter: joystick_enter {
+			label = "User joystick ENTER";
+			gpios = <&gpiog 8 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+		};
+		joystick_right: joystick_right {
+			label = "User joystick RIGHT";
+			gpios = <&gpioc 0 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+		};
+		joystick_up: joystick_up {
+			label = "User joystick UP";
+			gpios = <&gpioc 1 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+		};
+		joystick_down: joystick_down {
+			label = "User joystick DOWN";
+			gpios = <&gpioc 2 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+		};
+		joystick_left: joystick_left {
+			label = "User joystick LEFT";
+			gpios = <&gpioc 3 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+		};
+	};
+
+	aliases {
+		led0 = &led_1;
+		led1 = &led_2;
+		led2 = &led_3;
+		led3 = &led_4;
+		sw0 = &button;
+	};
+};
+
+&usart1 {
+	current-speed = <115200>;
+	status = "okay";
+};
+
+&usart2 {
+	current-speed = <115200>;
+	status = "okay";
+};
+
+&spi1 {
+	status = "okay";
+};
+
+&spi2 {
+	status = "okay";
+};
+
+&i2c1 {
+	status = "okay";
+	clock-frequency = <I2C_BITRATE_FAST>;
+};
+
+&i2c2 {
+	status = "okay";
+	clock-frequency = <I2C_BITRATE_FAST>;
+};
+
+&can1 {
+	/*
+	 * make sure CAN and USB are not enabled at the same time
+	 * because they share interrupts 19, 20 (stm32f103Xb.dtsi)
+	 * reference: RM0008 rev20 page 205
+	 */
+	status = "disabled";
+	bus-speed = <125000>;
+	sjw = <1>;
+	prop-seg = <0>;
+	phase-seg1 = <5>;
+	phase-seg2 = <6>;
+};
+
+&usb {
+	/*
+	 * make sure CAN and USB are not enabled at the same time
+	 * because they share interrupts 19, 20 (stm32f103Xb.dtsi)
+	 * reference: RM0008 rev20 page 205
+	 */
+	status = "okay";
```

Comment:
in the DTS, stm32f103Xe which includes stm32f103Xb, has USB interrupt <20>
( interrupt <19> is for _USB high priority or CAN TX_ interrupt on  XL-density devices )

Reply:
In stm32f103Xb.dtsi I see that USB uses interrupt 20.

Here is an extract from RM008 Table 63:
![RM008-table63](https://user-images.githubusercontent.com/5003901/86244719-ac6dd380-bba8-11ea-9e2d-bfee29e3bf55.png)

How to proceed? Shall I remove that comment?

Reply:
no need, that's fine as is.